[DOCS-314] Add sprint 2 pentest contributors

mjang-cobalt opened the pull request:
## Changelog

Sprint 2 not yet complete. Including 3 approved articles.

### Updated

| Page | Deploy Preview | Comment |
| ----- | ----- | ----- |
| Best Practices | [[Link]](https://deploy-preview-210--cobalt-docs.netlify.app/bestpractices/) | 3 articles, so far |

## Preview This Change

To see how this change looks in production, scroll down to **Deploy Preview**. Select the link that looks like `https://deploy-preview-<num>--cobalt-docs.netlify.app/`

## Variables

Help us support a “Write once, publish everywhere” single source of truth. If you see a line that looks like:

`{{ % asset-categories % }}`

You’ve found a [shortcode](https://gohugo.io/content-management/shortcodes/) that we include in multiple documents.

You’ll find the content of the shortcode in the following directory:

https://github.com/cobalthq/cobalt-product-public-docs/tree/main/layouts/shortcodes

That shortcode has the same base name as what you see in the PR, such as `asset-categories.html`.

mjang-cobalt commented:
@ana-dashuk-cobalt any idea why there's a gap between lines 3 and 4 of the table?
<img width="737" alt="Screen Shot 2022-10-18 at 11 47 38 AM" src="https://user-images.githubusercontent.com/84352037/196517921-e71234cb-3dc2-4b0d-af86-ea526ca84c0c.png">

ana-dashuk-cobalt commented:
> @ana-dashuk-cobalt any idea why there's a gap between lines 3 and 4 of the table? <img alt="Screen Shot 2022-10-18 at 11 47 38 AM" width="737" src="https://user-images.githubusercontent.com/84352037/196517921-e71234cb-3dc2-4b0d-af86-ea526ca84c0c.png">

@mjang-cobalt The same thing happened last time. I tested this several times. The table doesn't break when I set up most (or all) links in shortcodes in HTML format.

Example:

```
<a href="https://harshbothra.tech/" target="_blank">Harsh Bothra</a>
```

instead of

```
[Harsh Bothra](https://harshbothra.tech/)
```

When we use shortcodes inside other elements, they may break the element. The same thing happens with Note callouts.

mjang-cobalt commented:
Fixed with some copy/paste. (I moved some lines around, and the extra space always followed one of the lines) I'm also thinking, it's better to include the newest content atop the table.

<img width="812" alt="Screen Shot 2022-10-18 at 12 50 14 PM" src="https://user-images.githubusercontent.com/84352037/196530140-387ee88f-5768-4d53-94eb-d46ccc71eb9f.png">

ana-dashuk-cobalt commented on content/en/BestPractices/_index.md:
```diff
@@ -25,9 +25,11 @@ until we have enough articles to actually organize. -->
+|-----------------------------------------------------------------------|--------------|-----------------------|
+| [Prevent Insecure Design in SQL](./secure-design)                     | 2022-10-18   | {{% shashank %}}      |
+| [Protect Your APIs With Rate Limiting](./api-rate-limiting)           | 2022-10-18   | {{% payloadartist %}} |
+| [Prevent Server-Side Template Injections](./prevent-ssti)             | 2022-10-18   | {{% payloadartist %}} |
```
```suggestion
| [Prevent Server-Side Template Injections](./prevent-ssti)             | 2022-10-18   | {{% unstabl3 %}} |
```

ana-dashuk-cobalt commented:
> Fixed with some copy/paste. (I moved some lines around, and the extra space always followed one of the lines) I'm also thinking, it's better to include the newest content atop the table.
> 
> <img alt="Screen Shot 2022-10-18 at 12 50 14 PM" width="812" src="https://user-images.githubusercontent.com/84352037/196530140-387ee88f-5768-4d53-94eb-d46ccc71eb9f.png">

@mjang-cobalt The table looks great, but we're missing Shubham from the list.

ana-dashuk-cobalt reviewed (commented):
@mjang-cobalt The table looks broken again.

Should we go ahead and add HTML to shortcodes to avoid this problem in the future?

mjang-cobalt commented:
> Should we go ahead and add HTML to shortcodes to avoid this problem in the future?

I'm feeling dense. How would this help?

mjang-cobalt commented:
I have a hacky fix:

<img width="696" alt="Screen Shot 2022-10-19 at 10 09 16 AM" src="https://user-images.githubusercontent.com/84352037/196758579-c4fb1127-7f7d-4836-a8ae-83ebde4f614f.png">

ana-dashuk-cobalt reviewed (approved):
@mjang-cobalt Approving because there is no better solution at the moment. I tried adding HTML instead of Markdown to shortcodes but it didn't work (last time it did). Since it's not a stable solution, we'll stick to the workaround that you suggested.

Or I would suggest adding Markdown directly to the table, without shortcodes. Essentially these links will only live on a single page, so it's easy to copy them from one row to another.

ana-dashuk-cobalt commented on content/en/BestPractices/_index.md:
```diff
@@ -25,9 +25,18 @@ until we have enough articles to actually organize. -->
+
+| Article                                                               | Release Date | Author                |
+|-----------------------------------------------------------------------|--------------|-----------------------|
+| [Prevent Insecure Design in SQL](./secure-design)                     | 2022-10-18   | {{% shashank %}}      |
+| [Protect Your APIs With Rate Limiting](./api-rate-limiting)           | 2022-10-18   | {{% payloadartist %}} |
+| [Prevent Server-Side Template Injections](./prevent-ssti)             | 2022-10-18   | {{% unstabl3 %}}      |
```
```suggestion
| [Prevent Insecure Design in SQL](./secure-design)                     | 2022-10-20   | {{% shashank %}}      |
| [Protect Your APIs With Rate Limiting](./api-rate-limiting)           | 2022-10-20   | {{% payloadartist %}} |
| [Prevent Server-Side Template Injections](./prevent-ssti)             | 2022-10-20   | {{% unstabl3 %}}      |
```

ana-dashuk-cobalt replied:
@mjang-cobalt

mjang-cobalt commented:
> Or I would suggest adding Markdown directly to the table, without shortcodes. Essentially these links will only live on a single page, so it's easy to copy them from one row to another.

You may be right. 

I originally intended to include a sentence about each author in their articles, something like, "You can [read|learn] more about [authorname]'s work in [shortcode link]." But that doesn't work with current shortcodes. 

Maybe something simple, like include "Written by [shortcode author]." atop each page?

ana-dashuk-cobalt commented:
> Maybe something simple, like include "Written by [shortcode author]." atop each page?

I like this idea! @mjang-cobalt

mjang-cobalt commented on content/en/BestPractices/_index.md:
```diff
@@ -25,9 +25,18 @@ until we have enough articles to actually organize. -->
-| [Validate User Input](./input-validation)                      | 2022-08-19   | {{% payloadartist %}} |
-| [Prevent Security Misconfiguration](./prevent-security-misconfig)              | 2022-08-19   | {{% shashank %}}      |
-| [Protect Against Server-Side Request Forgery](./protect-against-ssrf) | 2022-08-19   | {{% harsh-bothra %}}  |
+### October 20, 2022
+
+| Article                                                               | Release Date | Author                |
+|-----------------------------------------------------------------------|--------------|-----------------------|
+| [Prevent Insecure Design in SQL](./secure-design)                     | 2022-10-18   | {{% shashank %}}      |
+| [Protect Your APIs With Rate Limiting](./api-rate-limiting)           | 2022-10-18   | {{% payloadartist %}} |
+| [Prevent Server-Side Template Injections](./prevent-ssti)             | 2022-10-18   | {{% unstabl3 %}}      |
```
```suggestion
### October 21, 2022

| Article                                                               | Release Date | Author                |
|-----------------------------------------------------------------------|--------------|-----------------------|
| [Prevent Insecure Design in SQL](./secure-design)                     | 2022-10-21   | {{% shashank %}}      |
| [Protect Your APIs With Rate Limiting](./api-rate-limiting)           | 2022-10-21   | {{% payloadartist %}} |
| [Prevent Server-Side Template Injections](./prevent-ssti)             | 2022-10-21   | {{% unstabl3 %}}      |
```
